TKeazirian.HTTPServer/Router/RoutesConfig.cs

```diff
@@ -1,30 +1,32 @@
+using HttpMethod = TKeazirian.HTTPServer.Response.HttpMethod;
+
 namespace TKeazirian.HTTPServer.Router;
 
 using Handler;
 
-public class RoutesConfig
+public static class RoutesConfig
 {
-    public readonly Dictionary<string, Handler> Routes;
-
-    public RoutesConfig()
+    public static Routes ConstructRoutes()
     {
-        Routes =
-            new Dictionary<string, Handler>
-            {
-                { "/echo_body", new EchoBodyHandler() },
-                { "/simple_get", new SimpleGetHandler() },
-                {
-                    "/simple_get_with_body", new SimpleGetHandler()
-                },
-                { "/redirect", new RedirectHandler() },
-                { "/head_request", new SimpleHeadHandler() },
-                { "/method_options", new SimpleOptionsHandler() },
-                { "/method_options2", new SimpleOptionsHandler2() },
-            };
-    }
+        Routes routes = new();
 
-    public RoutesConfig(Dictionary<string, Handler> routes)
-    {
-        Routes = routes;
+        routes.AddRoute(
+            new Route("/simple_get", new List<HttpMethod> { HttpMethod.GET }, new SimpleGetHandler())
+        );
+        routes.AddRoute(
+            new Route("/simple_get_with_body", new List<HttpMethod> { HttpMethod.GET }, new SimpleGetHandler())
+        );
+        routes.AddRoute(
+            new Route("/echo_body", new List<HttpMethod> { HttpMethod.POST }, new EchoBodyHandler()));
+        routes.AddRoute(
+            new Route("/redirect", new List<HttpMethod> { HttpMethod.GET }, new RedirectHandler()));
+        routes.AddRoute(
+            new Route("/head_request", new List<HttpMethod> { HttpMethod.GET }, new SimpleHeadHandler()));
+        routes.AddRoute(
+            new Route("/method_options", new List<HttpMethod> { HttpMethod.GET }, new SimpleOptionsHandler()));
+        routes.AddRoute(
+            new Route("/method_options2", new List<HttpMethod> { HttpMethod.GET, HttpMethod.PUT, HttpMethod.POST },
+                new SimpleOptionsHandler()));
```

Comment:
I notice there is not a 1:1 relationship between a path, and an handler. Instead you have two paths pointing to the same handler. This works, but for a large project with multiple endpoints, it could really come back to bite you.  Same goes the SimpleGetHandler reuse above.  

I know this is really a fault of how the acceptance test suite is coded, but I wanted to mention it here just so you are aware.

Reply:
Got it.  So normally (for a large project), the `/method_options` (for example) would have one handler, and then the `/method_options2` would have a separate handler?  One handler per one path, correct?  

There was [feedback](https://github.com/t-keazirian/HTTPServer/pull/27#discussion_r914256806) on by last PR where Travis was asking why I had two handlers - maybe he wasn't getting at only having one handler, but that was my impression.